plugins/optimization-detective/class-od-html-tag-processor.php

```diff
@@ -160,7 +179,31 @@ final class OD_HTML_Tag_Processor {
 	 * @param string $html HTML to process.
 	 */
 	public function __construct( string $html ) {
-		$this->processor = new WP_HTML_Tag_Processor( $html );
+
+		$processor = new class( $html ) extends WP_HTML_Tag_Processor {
+
+			/**
+			 * Appends HTML to the provided bookmark.
+			 *
+			 * @param string $bookmark Bookmark.
+			 * @param string $html     HTML to inject.
+			 * @return bool Whether the HTML was appended.
+			 */
+			public function append_html( string $bookmark, string $html ): bool {
+				if ( ! $this->has_bookmark( $bookmark ) ) {
+					return false;
+				}
+
+				$this->lexical_updates[] = new WP_HTML_Text_Replacement(
+					$this->bookmarks[ $bookmark ]->start,
+					0,
+					$html
```

Comment:
oh shoot, this may be a result of the change of signature of the span/range functions from `(start, end)` to `(start, length)`.

on WordPress 6.4 the span would be from `$this->bookmarks[ $bookmark ]->start` to `$this->bookmarks[ $bookmark ]->end` I think. I wouldn't understand why a `0` value here though would duplicate the string.

also it's suspiciously close to what would happen if we weren't to shadow the `$html` from line 183, but that seems impossible.

Reply:
Great! Easily fixed then in 03ad8f9f30311d55907addd7b8e6558cf33c48b3.